src/observable/dom/AjaxObservable.ts

```diff
@@ -417,7 +417,7 @@ export class AjaxResponse {
           this.response = JSON.parse(xhr.responseText || 'null');
         }
         break;
-      case 'xml':
```

Comment:
Indeed, this may be a breaking change - but there are 2 issues here:
- `xml` is not a valid responseType per the spec (see the links I gave in the PR description). So it seems that any server that returned `xml` as the responseType would not be compliant to the spec.
- Thus, `xml` is not accepted as a valid responseType by the TS 2.3 compiler (because the typings were made stricter, to disallow invalid responseTypes).

One approach to preserve compatibility may be to handle both the `xml` and `document` cases explicitly (via a type cast, etc).